java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/debugging/launch/NbLaunchRequestHandler.java

```diff
@@ -93,23 +93,18 @@
 
         activeLaunchHandler.preLaunch(launchArguments, context);
 
-        String filePath = (String)launchArguments.get("mainClass");
+        String filePath = (String)launchArguments.get("file");
+        String mainFilePath = (String)launchArguments.get("mainClass");
+        boolean preferProjActions = true; // True when we prefer project actions to the current (main) file actions.
+        if (filePath == null || mainFilePath != null) {
+            // main overides the current file
+            preferProjActions = false;
+            filePath = mainFilePath;
+        }
         FileObject file = null;
         File nativeImageFile = null;
         if (!isNative) {
-            File ioFile = null;
-            if (filePath != null) {
-                ioFile = new File(filePath);
-                if (!ioFile.exists()) {
-                    try {
-                        URI uri = new URI(filePath);
-                        ioFile = Utilities.toFile(uri);
-                    } catch (URISyntaxException ex) {
-                        // Not a valid file
-                    }
-                }
-            }
-            file = ioFile != null ? FileUtil.toFileObject(ioFile) : null;
+            file = getFileObject(filePath);
```

Comment:
I didn't run the code, but ... if `mainClass` is defined, then `filePath` now contains main class' FQN - so it can't be converted to FileObject ?